include/exadg/solvers_and_preconditioners/preconditioners/block_jacobi_preconditioner.h

```diff
@@ -58,11 +61,12 @@ class BlockJacobiPreconditioner : public PreconditionerBase<typename Operator::v
   void
   vmult(VectorType & dst, VectorType const & src) const
   {
-    underlying_operator.apply_inverse_block_diagonal(dst, src);
+    underlying_operator.apply_inverse_block_diagonal(block_matrix, dst, src);
   }
 
 private:
   Operator const & underlying_operator;
+  BlockMatrix      block_matrix;
```

Comment:
Here, it looks like the `BlockJacobiPreconditioner` always uses a matrix-based implementation technique. In case a matrix-free application of the inverse block "matrices" is chosen, `block_matrix` is probably not used. We should make sure how to make this intuitively understandable when looking at the implementation of `BlockJacobiPreconditioner`.